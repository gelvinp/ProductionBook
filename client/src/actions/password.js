export const SET_PASSWORD = 'SET_PASSWORD'

export function setPassword(pass) {
  return { type: SET_PASSWORD, pass: pass }
}
