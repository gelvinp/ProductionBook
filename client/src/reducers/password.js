import { SET_PASSWORD } from '../actions/password.js'

export function password(state = '', action) {
  switch (action.type) {
    case SET_PASSWORD:
      return action.pass
    default:
      return state
  }
}
