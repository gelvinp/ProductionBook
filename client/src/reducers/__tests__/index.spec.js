import rootReducer from '../index.js'
import { SET_PASSWORD } from '../../actions/password.js'

describe('root reducer', () => {
  it('should reset with password -1', () => {
    expect(
      rootReducer({ test: true }, { type: SET_PASSWORD, pass: -1 })
    ).toMatchSnapshot()
  })
})
