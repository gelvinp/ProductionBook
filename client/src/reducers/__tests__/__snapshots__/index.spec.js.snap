// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root reducer should reset with password -1 1`] = `
Object {
  "documents": Object {
    "selectedDocument": Object {
      "section": -1,
      "uuid": -1,
    },
  },
  "modals": Object {
    "document": Object {},
    "section": -1,
    "uploadOpen": false,
  },
  "password": -1,
  "sections": Object {},
}
`;
